plugins/modules/foreman_operating_system.py

```diff
@@ -139,47 +166,37 @@
 
 RETURN = ''' # '''
 
-try:
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_entities,
-        find_entities_by_name,
-        find_operating_system_by_title,
-        naildown_entity,
-        naildown_entity_state,
-        sanitize_entity_dict,
-        OperatingSystemParameter,
-    )
 
-    from nailgun.entities import (
-        Architecture,
-        Media,
-        OperatingSystem,
-        PartitionTable,
-        ProvisioningTemplate,
-    )
-except ImportError:
-    pass
+from ansible.module_utils.foreman_helper import ForemanEntityApypieAnsibleModule, parameter_value_to_str
 
-from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
 
 # This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-    'description': 'description',
-    'family': 'family',
-    'major': 'major',
-    'minor': 'minor',
-    'release_name': 'release_name',
-    'architectures': 'architecture',
-    'media': 'medium',
-    'ptables': 'ptable',
-    'provisioning_templates': 'provisioning_template',
-    'password_hash': 'password_hash',
+entity_spec = {
+    'id': {},
```

Comment:
I liked to have the id implicit :p

Reply:
I prefer explicit, but could be convinced.
Also for the cases, where we rename things like `from_ip -> `from`, we need to add both here in order not to delete it from the current_entity. At least, i could not think of an easier way to encode alias names.

Reply:
I think my argument for implicit is "we carry only user visible or better user input in that list"

Reply:
We could stop this renaming (from/from_ip) madness. It's a remainder from nailgun which could not use python reserved words. You can use them in dicts tho. And an alias in the argument spec is cheap.

Reply:
We also need to specify all other properties, that are different in the api answer.